clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/LimitAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package com.clevertap.android.sdk.inapp.evaluation
+
+import com.clevertap.android.sdk.Constants
+import org.json.JSONObject
+
+/**
+ * Enumeration representing different types of limits for in-app notifications.
+ *
+ * Each enum value corresponds to a specific type of limit that can be applied, such as "ever," "session,"
+ * "seconds," "minutes," "hours," "days," "weeks," "onEvery," and "onExactly."
+ *
+ * @property type The string representation of the limit type.
+ */
+enum class LimitType(val type: String) {
```

Comment:
https://stackoverflow.com/questions/49169086/sealed-class-vs-enum-when-using-associated-type
Is using sealed class better in this case? I am asking since it would be exhaustive when statement. Please go through this so we can decide.